image/mkosi/secure-boot/generate_nvram_vars.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+BASE_DIR=$(realpath "${SCRIPT_DIR}/..")
+PKI=${BASE_DIR}/pki
+
+generate_ovmf_vars () {
+    # should work in theory but seems to fail on ubuntu
+    # TODO: fix this or remove it
+    # generate ovmf vars file using https://github.com/rhuefi/qemu-ovmf-secureboot
+    # Uses https://github.com/tianocore/edk2/blob/master/OvmfPkg/EnrollDefaultKeys/EnrollDefaultKeys.c (EnrollDefaultKeys.efi)
+    # to deploy our cert as PK, KEK and db.
+    if test -f "${PKI}/image_SB_VARS.fd"; then
+        echo "OVMF vars already generated."
+        return
+    fi
+    mkdir -p "${PKI}"
+
+    sed \
+        -e 's/^-----BEGIN CERTIFICATE-----$/4e32566d-8e9e-4f52-81d3-5bb9715f9727:/' \
+        -e '/^-----END CERTIFICATE-----$/d' \
+        ${PKI}/db.crt > ${PKI}/PkKek1.oemstr
```

Comment:
??? 

If you keep this (fix for Ubuntu would be great :P) please comment what this does. Also, I could imagine that this here is why this fails on Ubuntu (different PKI?).

Reply:
This is actually required for the ovmf vars generator regardless of the platform. The magic value is used as an efivar and used by one of the efi binaries to detect the wanted secure boot key. Will document. And maybe remove if this does not work.